releasenotes/notes/sql-obfuscation-config-options-d9dfbc9ca568c63f.yaml

```diff
@@ -0,0 +1,15 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    dbm: add SQL obfuscation options to give customer more control over how SQL is obfuscated and normalized.
+    - `RemoveSpaceBetweenParentheses` - remove spaces between parentheses. This option is only valid when ObfuscationMode is "obfuscate_and_normalize".
+    - `KeepNull` - disable obfuscating null values with ?. This option is only valid when ObfuscationMode is "obfuscate_only" or "obfuscate_and_normalize".
+    - `KeepBoolean` - disable obfuscating boolean values with ?. This option is only valid when ObfuscationMode is "obfuscate_only" or "obfuscate_and_normalize".
+    - `KeepPositionalParameter` - disable obfuscating positional parameters with ?. This option is only valid when ObfuscationMode is "obfuscate_only" or "obfuscate_and_normalize".
```

Comment:
Can we document how these options can be configured by the user? e.g `RemoveSpaceBetweenParentheses` is `remove_space_between_parentheses` in the yaml, we should only focus on the customer facing syntax. Also is there a way to configure these options via env variables?

Reply:
Two small nits:
1. Changelog format is RST, so needs double backticks for monospace font.
2. Config structure uses snake-cased json keys. Would it make sense to use the same spelling here in the user visible docs? Or will camel-case work as well?

Reply:
@ahmed-mez These options will be used by the DBM python integrations in `integrations-core` via the c binding `obfuscate_sql_with_metadata`. My plan is to update the `spec.yaml` in DBM integrations once the datadog-agent change is merged. The options are not configurable via env vars.